nifpga/session.py

```diff
@@ -592,44 +615,55 @@ def _convert_written_value_to_fxp_representation(self, data):
             fxp_representation = self._integer_twos_comp(fxp_representation)
 
         if self._overflow_enabled:
-            self._check_overflow_attribute_is_set()
-            if self.overflow:
+            if overflow:
                 fxp_representation += 2**(self._word_length)
+
         return fxp_representation
 
+    def _validate_and_parse_user_input(self, user_input):
+        overflow = None
+        data = None
+        if self._overflow_enabled:
+            try:
+                (overflow, data) = user_input
+            except TypeError:
+                """ If the user does not input any overflow status, eat the
+                exception and use default value of False. """
+                overflow = False
+                data = user_input
+            assert isinstance(overflow, bool)
+        else:
+            data = user_input
+        assert isinstance(data, Number)
+        return (overflow, data)
+
     def _convert_value_to_fxp(self, data):
         calculated_fxp = Decimal(data) / Decimal(self._delta)
         fxp_representation = int(calculated_fxp)
-        """ If the result of the division is not an integer we lost some of
-        the input data warn the user that we had to coerce the value to the
-        nearest fixed point representation. """
+        """ If the result of the division is not an integer, we lost some of
+        the input data. In this case we warn the user that we had to coerce the
+        value to the nearest fixed point representation. """
```

Comment:
Does "round to nearest" correspond to the default Decimal context? Is there any uncertainty about the default context (should you specify an explicit context somewhere)?